internal/driver/ipmi/ipmi.go

```diff
@@ -162,3 +162,16 @@ func (s *server) SetBootdev(dev string) error {
 	}
 	return s.ipmitool("chassis", "bootdev", dev, "options=persistent")
 }
+
+// Get the server's power status as a string.
+func (s *server) GetPowerStatus() (string, error) {
+	var status string
+	var errormsg error
+	s.RunInServer(func() {
+		out, err := s.info.ipmitool("chassis", "power", "status").Output()
+		output := string(out)
+		status = output
+		errormsg = err
```

Comment:
A few tips re: making this more concise:

* Declare `out` stand-alone, so you don't have to declare separate err and errormsg variables:

```go
var out []byte
out, err = ... // Note = rather than :=. err is now the variable declared in the parent scope, instead of errormsg.
```

* use named return values

```go
func (s *server) GetPowerStatus() (status string, err error) {
        // no need for separate variable declarations
        s.RunInServer(...)
}
```

* Get rid of the intermediate `output` variable.

```go
status = string(out)
```

The whole thing ends up looking like:

```go
func (s *server) GetPowerStatus() (status string, err error) {
        s.RunInServer(func() {
                var out []byte
                out, err = s.info...
                status = string(out)
        })
}

Reply:
Getting used to return syntax, so at the end of my GetPowerStatus() function I just put a `return` and it will return both status and err?